src/OpenLoco/src/Audio/Audio.cpp

```diff
@@ -1027,68 +1028,93 @@ namespace OpenLoco::Audio
+    static std::vector<uint8_t> makeAllMusicPlaylist()
     {
+        std::vector<uint8_t> playlist(kNumMusicTracks);
+        std::iota(playlist.begin(), playlist.end(), 0);
+        return playlist;
+    }
+
+    static std::vector<uint8_t> makeCurrentEraPlaylist()
+    {
+        auto playlist = std::vector<uint8_t>();
+        auto currentYear = getCurrentYear();
+
         for (auto i = 0; i < kNumMusicTracks; i++)
         {
-            if (i != excludeTrack)
+            const auto& mi = kMusicInfo[i];
+            if (currentYear >= mi.startYear && currentYear <= mi.endYear)
             {
                 playlist.push_back(i);
             }
         }
+
+        return playlist;
     }
 
-    static void addCurrentEraSongsToPlaylist(int32_t excludeTrack, std::vector<uint8_t>& playlist)
+    static std::vector<uint8_t> makeCustomSelectionPlaylist()
     {
-        auto currentYear = getCurrentYear();
+        auto playlist = std::vector<uint8_t>();
+
+        const auto& cfg = Config::get().old;
         for (auto i = 0; i < kNumMusicTracks; i++)
         {
-            const auto& mi = kMusicInfo[i];
-            if (currentYear >= mi.startYear && currentYear <= mi.endYear)
+            if (cfg.enabledMusic[i] & 1)
             {
-                if (i != excludeTrack)
-                {
-                    playlist.push_back(i);
-                }
+                playlist.push_back(i);
             }
         }
+
+        return playlist;
     }
 
-    static int32_t chooseNextMusicTrack(int32_t excludeTrack)
+    std::vector<uint8_t> makeSelectedPlaylist()
     {
         using MusicPlaylistType = Config::MusicPlaylistType;
 
-        static std::vector<uint8_t> playlist;
-        playlist.clear();
-
-        const auto& cfg = Config::get().old;
-        switch (cfg.musicPlaylist)
+        switch (Config::get().old.musicPlaylist)
         {
             case MusicPlaylistType::currentEra:
-                addCurrentEraSongsToPlaylist(excludeTrack, playlist);
-                break;
+                return makeCurrentEraPlaylist();
+
             case MusicPlaylistType::all:
-                addAllSongsToPlaylist(excludeTrack, playlist);
-                break;
+                return makeAllMusicPlaylist();
+
             case MusicPlaylistType::custom:
-                for (auto i = 0; i < kNumMusicTracks; i++)
-                {
-                    if (i != excludeTrack && (cfg.enabledMusic[i] & 1))
-                    {
-                        playlist.push_back(i);
-                    }
-                }
-                break;
+                return makeCustomSelectionPlaylist();
         }
 
+        throw Exception::RuntimeError("Invalid MusicPlaylistType");
+    }
+
+    static int32_t chooseNextMusicTrack()
+    {
+        using MusicPlaylistType = Config::MusicPlaylistType;
+
+        static std::vector<uint8_t> playlist;
+        playlist = makeSelectedPlaylist();
```

Comment:
The static does nothing here anymore.
```suggestion
        auto playlist = makeSelectedPlaylist();
```